disnake/asset.py

```diff
@@ -191,14 +194,14 @@ class Asset(AssetMixin):
 
     BASE = "https://cdn.discordapp.com"
 
-    def __init__(self, state, *, url: str, key: str, animated: bool = False) -> None:
-        self._state = state
-        self._url = url
-        self._animated = animated
-        self._key = key
+    def __init__(self, state: AnyState, *, url: str, key: str, animated: bool = False) -> None:
+        self._state: AnyState = state
+        self._url: str = url
+        self._animated: bool = animated
+        self._key: str = key
 
     @classmethod
-    def _from_default_avatar(cls, state, index: int) -> Asset:
+    def _from_default_avatar(cls: Type[T], state: AnyState, index: int) -> T:
```

Comment:
classmethods like these were changed from `(cls: Type[T]) -> T` to `(cls) -> Self` a while ago, importing `typing_extensions.Self` in a type-checking-only block (as we still don't explicitly require it). Would you mind updating these methods to use that as well?

Reply:
Sure!

Reply:
fdec647f0b418c42cb3be3641c0a0eaff4ce9662 ~~forgot I pushed it~~